### DISCUSSION TITLE
Picture uploads appear to being truncated

### DISCUSSION THREAD

**Discussion by lracselwonk** [easytarget/esp32-cam-webserver#251] [Common Issues] [answered]
I am experiencing this problem with a time lapse project whereby images are taken every 30 seconds then uploaded to my server. It would appear that the problem is worse during bright light but it can occur under any circumstances. The camera settings are the sefault values and while I have tried changing various settings nothing seems to solve the problem. Any help would be appreciated.

![badpics](https://user-images.githubusercontent.com/90358984/165952630-796b4b71-dd23-4d2e-969c-df2beaeac96d.jpg)

**Comment by easytarget** [comment, marked as answer]
You will see truncated images like this if you set both the quality level very high and the framesize very high (HD quality+)
- The 'Quality' control is essentially the 'compression level' control.
- Large HQ images are very large.. at some point they will overflow the buffer (psram) and get truncated.
- Make sure that whatever combo you are using works properly for still images in the web UI.

If that does not help:
* Double-check psram is enabled (you can see this in the status page)

if it is:
* Try changing the xclock speed in the config. With many modules (including one of the four I own) you can get partial images at the default 20MHz speed, try lowering it in stages.

I have seen issues exactly like this, truncated images, so I understand what you are seeing. It is related to either errors obtaining the whole image (xclock is your defence against this), or not having enough free memory for the frame.

**Reply by lracselwonk** [reply]
Thanks, I'll try adjusting the quality settings.  What exactly is the xclock setting, CPU frequency?

**Reply by easytarget** [reply]
The ESP32 uses a serial bus (SCCB) to communicate with the camera module, the xclk setting controls the speed of that bus. 

Communications errors on that bus show up as initialisation failures in bad cases. But more commonly you see glitches in images and/or truncated frames.

Lowering the xclk can clear the errors, but at the expense of slower comms (and hence frame rates, etc).

A lot of budget camera modules seem to have issues with this.

**Reply by lracselwonk** [reply]
Thanks that seems to have solved problem. I've set xclk_freq_hz = 15000000 and no more truncated or partial images. I also tried reducing quality to SXGA and XGA as well but there was no problem when I went with UXGA and reduced clock speed. You are a saviour!
